.github/workflows/check.yml

```diff
@@ -4,23 +4,23 @@ jobs:
   check:
     strategy:
       matrix:
-        os:
-          [
-            '{"runs-on": "ubuntu-latest"}',
-            '{"runs-on": "macos-latest"}',
-            '{"runs-on": "ubuntu-latest", "container": "ubuntu:18.04"}',
-            '{"runs-on": "windows-latest"}',
-          ]
-    runs-on: ${{ fromJSON(matrix.os).runs-on }}
-    container: ${{ fromJSON(matrix.os).container }}
+        system:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        include:
+          - system: ubuntu-18.04
+            container: ubuntu:18.04    
+    runs-on: matrix.system
+    container: matrix.container
```

Comment:
presumably fails as this is undefined in 3/4 cases?

Reply:
curious... GHA seems to have defaults which get propagated fine.

Reply:
Actually, keys with undefined values act like unspecified keys. It failed because I committed a copy-paste mistake: missing `${{ }}` for the 14th and 15th lines.